src/py42/services/cases.py

```diff
@@ -0,0 +1,194 @@
+
+from py42.services import BaseService
+from py42.services.util import get_all_pages
+from py42.settings import items_per_page
+
+
+class CasesService(BaseService):
+    """`Rest documentation https://default-cases.core-int.cloud.code42.com/swagger-ui.html#/Cases`__ ."""
+
+    _uri_prefix = "/api/v1/case"
+
+    def __init__(self, connection):
+        super(CasesService, self).__init__(connection)
+
+    def create(self, name, subject, assignee, description, findings):
+        """Creates a new case.
+
+        Args:
+            name (str): Name of the case.
+            subject (str): A subject of the case.
+            assignee (str): User UID of the assignee.
+            description (str): Description of the case
+            findings (str): Observations of the case.
+
+        Returns
+            :class:`py42.response.Py42Response`
+        """
+        data = {
+            "assignee": assignee,
+            "description": description,
+            "findings": findings,
+            "name": name,
+            "subject": subject,
+        }
+        return self._connection.post(self._uri_prefix, data)
+
+    def get_page(
+        self,
+        name=None,
+        status=None,
+        created_at=None,
+        updated_at=None,
+        subject=None,
+        assignee=None,
+        page_number=1,
+        page_size=items_per_page,
+        sort_direction="asc",
+        sort_key="number",
+        **kwargs
+    ):
+        """Gets an individual page of cases.
+
+        Args:
+            name (str, optional): Filter results by case name, matches partial names. Defaults to None.
+            status (str, optional): Filter results by case status, "OPEN" or "CLOSED". Defaults to None.
+            created_at (str, optional): Filter results by case creation time range, format ISO interval. Defaults to None. e.g 2020-08-31T11:00:00Z/2020-09-01T15:30:00Z
+            updated_at (str, optional): Filter results by last updated time range, format ISO interval. Defaults to None. e.g 2020-08-31T11:00:00Z/2020-09-01T15:30:00Z
+            subject (str, optional): Filter results based on User UID of a subject of a case. Defaults to None.
+            assignee (str, optional): Filter results based on User UID of an assignee of a case. Defaults to None.
+            page_number (int, optional): Page number of the results. Defaults to 1.
+            page_size (int, optional): Number of results to return per page. Defaults to 500.
+            sort_direction (str, optional): The direction on which to sort the response, based on the corresponding sort key. "asc" or "desc" Defaults to "asc".
+            sort_key (str, optional): Values on which the response will be sorted. Defaults to "number". Available options are
+             name, number, createdAt, updatedAt, status, assigneeUsername, subjectUsername
+
+        Returns:
+            :class:`py42.response.Py42Response`
+        """
+        param_dict = OrderedDict()
```

Comment:
the `get` method has an optional argument called `params` that you can pass a dict, and it will automatically handle creating the query string. It will also automatically exclude any item that has `None` assigned as its value.

Reply:
Changed.